pyretailscience/plots/styles/styling_context.py

```diff
@@ -0,0 +1,123 @@
+
+        # Built-in font registry - maps names to bundled font files
+        self._builtin_fonts = {
+            "poppins_bold": f"{ASSETS_PATH}/fonts/Poppins-Bold.ttf",
+            "poppins_semi_bold": f"{ASSETS_PATH}/fonts/Poppins-SemiBold.ttf",
+            "poppins_regular": f"{ASSETS_PATH}/fonts/Poppins-Regular.ttf",
+            "poppins_medium": f"{ASSETS_PATH}/fonts/Poppins-Medium.ttf",
+            "poppins_light_italic": f"{ASSETS_PATH}/fonts/Poppins-LightItalic.ttf",
+        }
+
+    def get_font_properties(self, font_name: str) -> fm.FontProperties:
+        """Get matplotlib FontProperties with flexible font resolution.
+
+        This method resolves fonts in the following priority order:
+        1. Check cache for previously loaded font
+        2. Try to load from bundled fonts if font_name matches a built-in font
+        3. Try to load as a system font family name
+        4. Fall back to matplotlib's default font
+
+        Args:
+            font_name (str): The name of the font to load. Can be:
+                - A built-in font name (e.g., 'poppins_regular')
+                - A system font family name (e.g., 'Arial', 'Times New Roman')
+
+        Returns:
+            fm.FontProperties: A matplotlib FontProperties object that can be used
+                for text rendering. Always returns a valid FontProperties object,
+                falling back to the default font if the requested font cannot be loaded.
+        """
+        if font_name in self._font_cache:
+            return self._font_cache[font_name]
+
+        # Try to load built-in bundled fonts first
+        if font_name in self._builtin_fonts:
+            try:
+                font_props = fm.FontProperties(fname=self._builtin_fonts[font_name])
+                logger.debug("Successfully loaded built-in font: %s", font_name)
+                self._font_cache[font_name] = font_props
+            except (OSError, RuntimeError) as e:
+                logger.warning(
+                    "Failed to load built-in font '%s' from %s: %s. Trying system fonts.",
+                    font_name,
+                    self._builtin_fonts[font_name],
+                    e,
+                )
+            else:
+                return font_props
+
+        # Try to load as system font family
+        try:
+            font_props = fm.FontProperties(family=font_name)
+            logger.debug("Successfully loaded system font family: %s", font_name)
+            self._font_cache[font_name] = font_props
+        except (OSError, RuntimeError) as e:
+            logger.warning("Failed to load system font family '%s': %s. Using default font.", font_name, e)
+        else:
+            return font_props
+        # Fall back to default font
+        font_props = fm.FontProperties()
+        logger.info("Using default font as fallback for '%s'", font_name)
+        self._font_cache[font_name] = font_props
+        return font_props
```

Comment:
Let's get rid of the fall back. If you specify you want a font and you don't get it I think it should just error.
```suggestion
        # Try to load as system font family
        try:
        font_props = fm.FontProperties(family=font_name)
        self._font_cache[font_name] = font_props
        return font_props
```